src/main/java/jenkins/plugins/git/BuiltRevisionMap.java

```diff
@@ -0,0 +1,154 @@
+package jenkins.plugins.git;
+
+import hudson.BulkChange;
+import hudson.Functions;
+import hudson.XmlFile;
+import hudson.model.Action;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.model.Saveable;
+import hudson.model.listeners.SaveableListener;
+import hudson.plugins.git.Branch;
+import hudson.plugins.git.util.BuildData;
+import org.eclipse.jgit.lib.ObjectId;
+import org.kohsuke.stapler.export.ExportedBean;
+
+import javax.annotation.CheckForNull;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static hudson.Util.fixNull;
+
+/**
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+@ExportedBean
+public class BuiltRevisionMap implements Action, Saveable {
+
+    public static final String FILE = BuiltRevisionMap.class.getName() + ".xml";
```

Comment:
In my new plugin i decided to use .runtime.xml file mask for all runtime data to not track constantly changed data (should be useful for example for jobConfigHistory).